core/model/src/main/java/org/eclipse/rdf4j/model/impl/LinkedHashModel.java

```diff
@@ -200,7 +200,9 @@ public Iterator iterator() {
 
 	@Override
 	public boolean contains(Resource subj, IRI pred, Value obj, Resource... contexts) {
-		return matchPattern(subj, pred, obj, contexts).hasNext();
+		ModelIterator modelIterator = matchPattern(subj, pred, obj, contexts);
+		Thread.yield();
+		return modelIterator.hasNext();
```

Comment:
I assume this is here merely for testing/demonstration purposes, and not intended to be a fix for anything?

Reply:
Just increases the likelihood of the error occurring.